### PR TITLE
[CIS-967] Fix ContainerStackView spacing issue

### PR DESCRIPTION
**Pull request by skorepak**
## This PR:
- fixes issue where spacing didn't work when the subview of container stack view was hidden twice before showing. 
- Add snapshot test for this case

## Links: 
https://stream-io.atlassian.net/browse/CIS-967

### CODE REVIEW

**Review by VojtaStavik — APPROVED** [review]

File: Sources/StreamChatUI/CommonViews/ContainerStackView.swift [hunk 1]
```diff
@@ -496,13 +496,15 @@ extension UILayoutPriority {
 
 extension NSLayoutConstraint {
     func setTemporaryConstant(_ value: CGFloat) {
+        guard originalConstant == nil else { return }
```
Comment (evsaev) [inline Sources/StreamChatUI/CommonViews/ContainerStackView.swift:499]: hm, the observer shouldn't even be invoked when there's an attempt to hide already hidden subview 🤔

```swift
hidingObserversByView[subview] = subview
            .observe(\.isHidden, options: [.new]) { [weak self] (view, isHiddenChange) in
                if isHiddenChange.newValue == true {
                    self?.hideArrangedSubview(view)
                } else {
                    self?.showArrangedSubview(view)
                }
            }
```

Comment (skorepak) [reply Sources/StreamChatUI/CommonViews/ContainerStackView.swift:499]: That's what I would expect! But the test I added is failing without this fix 😞

Comment (evsaev) [reply Sources/StreamChatUI/CommonViews/ContainerStackView.swift:499]: hm, how about
```swift
.observe(\.isHidden, options: [.new, .old]) { [weak self] (view, isHiddenChange) in
    guard isHiddenChange.oldValue != isHiddenChange.newValue else { return }
    ...
}
```

Comment (skorepak) [reply Sources/StreamChatUI/CommonViews/ContainerStackView.swift:499]: I was testing this and I think the logic failed sometimes. 
The `show/hideArrangedSubview` has some logic inside, which may result in calling the `hideArrangedSubview` method, but not calling the `setTemporaryConstant` method inside. Therefore I fixed it on the lowest level possible to remove any other logic involved on the higher level

**Review by evsaev — APPROVED** [review]
🚢

**Pull request merged**